graph/graphtest/graphtest.go

```diff
@@ -526,6 +528,65 @@ func TestLoadTypedQuads(t testing.TB, gen DatabaseFunc, conf *Config) {
 	require.Equal(t, int64(7), qs.Size(), "Unexpected quadstore size")
 }
 
+// Came across a bug where a time.Time quad returned by iterator.NewAnd could not be deleted even though it exists.
+// dennwc said that this has something to do with protobuf encoding dropping timezone info which is used in hash/node id generation.
+func TestReadTimeQuadsViaIterator(t testing.TB, gen DatabaseFunc, conf *Config) {
+	qs, opts, closer := gen(t)
+	defer closer()
+
+	w := MakeWriter(t, qs, opts)
+	now := time.Now()
+	subject := quad.String("timequad")
+	values := []quad.Quad{
+		{subject, quad.String("has_unixtime"), quad.Int(now.UnixNano()), nil},
+		{subject, quad.String("has_time"), quad.Time(now), nil},
+	}
+	err := w.AddQuadSet(values)
+	require.Nil(t, err)
+
+	for _, v := range values {
```

Comment:
In the check, when I was iterating through the values, I was passing two `QuadIterator`'s to `NewAnd`, one for subject and one for predicate so there should have been only one result returned.  I've moved (am moving) the read / equality check to `TestLoadTypedQuads` though.